src/main/d3utils.js

```diff
@@ -34,17 +34,22 @@ function formatRange(viewSize: number): any {
 }
 
 /**
- * Set attributes on an element. This can be used in place of, e.g.
- * d3.select(div).attr({width, height});
+ * Sizes a canvas appropriately for this device.
  */
-function setAttributes(el: Element, attrs: {[key:string]: any}) {
-  for (var k in attrs) {
-    el.setAttribute(k, attrs[k]);
+function sizeCanvas(el: HTMLCanvasElement, width: number, height: number) {
+  var ratio = window.devicePixelRatio;
+  el.width = width * ratio;
+  el.height = height * ratio;
+  el.style.width = width + 'px';
```

Comment:
Notice that they're set to different things. On a retina display, for example, `ratio=2`. So you might have `canvas.width=800` whereas `canvas.style.width="400px"`. This ensures that there are the right number of pixels to match the resolution of the screen.